File: RKLM_Reference/NumericsFundamentals/Discretization/time_discretization.c
#include "userdata.h"

/* ================================================================================ */

void set_time_integrator_parameters(User_Data *ud)
{
    
    /* Versions prior to Mar 4, 2018 included settings for more integrators 
    switch (ud->time_integrator) {
        case SI_MIDPT:
            ud->tips.dt_frac         = 0.5;
            ud->tips.flux_frac[0][0] = 0.0;
            ud->tips.flux_frac[0][1] = 1.0;
            ud->tips.flux_frac[1][0] = 0.5;
            ud->tips.flux_frac[1][1] = 0.5;
            for (int k=2; k<NO_OF_RK_STAGES; k++) {
                ud->tips.flux_frac[k][1] = 1e10;
                ud->tips.flux_frac[k][1] = 1e10;
            }
            ud->tips.update_frac[0]  = 0.5;
            ud->tips.update_frac[1]  = 1.0;
            for (int k=2; k<NO_OF_RK_STAGES; k++) {
                ud->tips.update_frac[k] = 1e10;
            }
            ud->tips.multiD_updt     = WRONG;
            break;
            
        default:
            assert(0);
            break;
    }
     */ 
    switch (ud->advec_time_integrator) {
      case EXPL_MIDPT:
            ud->tips.dt_frac         = 0.5;
            ud->tips.flux_frac[0][0] = 0.0;
            ud->tips.flux_frac[0][1] = 1.0;
            ud->tips.flux_frac[1][0] = -0.5;
            ud->tips.flux_frac[1][1] = 1.0;
            for (int k=2; k<NO_OF_RK_STAGES; k++) {
                ud->tips.flux_frac[k][1] = 1e10;
                ud->tips.flux_frac[k][1] = 1e10;
            }
            ud->tips.update_frac[0]  = 0.5;
            ud->tips.update_frac[1]  = 1.0;
            for (int k=2; k<NO_OF_RK_STAGES; k++) {
                ud->tips.update_frac[k] = 1e10;
            }
            ud->tips.multiD_updt     = CORRECT;
            break;
            
        case HEUN:
            ud->tips.dt_frac         = 0.5;
            ud->tips.flux_frac[0][0] = 0.0;
            ud->tips.flux_frac[0][1] = 1.0;
            ud->tips.flux_frac[1][0] = -0.5;
            ud->tips.flux_frac[1][1] = 0.5;
            for (int k=2; k<NO_OF_RK_STAGES; k++) {
                ud->tips.flux_frac[k][1] = 1e10;
                ud->tips.flux_frac[k][1] = 1e10;
            }
            ud->tips.update_frac[0]  = 1.0;
            ud->tips.update_frac[1]  = 1.0;
            for (int k=2; k<NO_OF_RK_STAGES; k++) {
                ud->tips.update_frac[k] = 1e10;
            }
            ud->tips.multiD_updt     = CORRECT;
            break;
            
        default: /* STRANG */
            ud->tips.dt_frac         = 0.5;
            ud->tips.flux_frac[0][0] = 0.0;
            ud->tips.flux_frac[0][1] = 1.0;
            ud->tips.flux_frac[1][0] = 0.5;
            ud->tips.flux_frac[1][1] = 0.5;
            for (int k=2; k<NO_OF_RK_STAGES; k++) {
                ud->tips.flux_frac[k][1] = 1e10;
                ud->tips.flux_frac[k][1] = 1e10;
            }
            ud->tips.update_frac[0]  = 0.5;
            ud->tips.update_frac[1]  = 1.0;
            for (int k=2; k<NO_OF_RK_STAGES; k++) {
                ud->tips.update_frac[k] = 1e10;
            }
            ud->tips.multiD_updt     = WRONG;
            break;
    }
}

File: RKLM_Python/discretization/time_discretization.py
# import numpy as np
# from management.enumerator import TimeIntegrator

# def SetTimeIntegratorParameters(ud):
#     if ud.advec_time_integrator == TimeIntegrator.EXPL_MIDPT:
#         ud.tips.dt_frac = 0.5
#         ud.tips.flux_frac[0][0] = 0.0
#         ud.tips.flux_frac[0][1] = 1.0
#         ud.tips.flux_frac[1][0] = -0.5
#         ud.tips.flux_frac[1][1] = 1.0

#         for k in range(2,ud.tips.NO_OF_RK_STAGES):
#             ud.tips.flux_frac[k][1] = 1e10

#         ud.tips.update_frac[0] = 0.5
#         ud.tips.update_frac[1] = 1.0

#         for k in range(2,ud.tips.NO_OF_RK_STAGES):
#             ud.tips.update_frac[k] = 1e10
        
#         ud.tips.multiD_updt = True
    
#     elif ud.advec_time_integrator == TimeIntegrator.HUEN:
#         ud.tips.dt_frac = 0.5
#         ud.tips.flux_frac[0][0] = 0.0
#         ud.tips.flux_frac[0][1] = 1.0
#         ud.tips.flux_frac[1][0] = -0.5
#         ud.tips.flux_frac[1][1] = 0.5

#         for k in range(ud.tips.NO_OF_RK_STAGES):
#             ud.tips.flux_frac[k,:] = 1e10

#         ud.tips.update_frac[0] = 1.0
#         ud.tips.update_frac[1] = 1.0

#         for k in range(ud.tips.NO_OF_RK_STAGES):
#             ud.tips.update_frac[k] = 1e10
        
#         ud.tips.multiD_updt = True
        
#     else:
#         ud.tips.dt_frac = 0.5
#         ud.tips.flux_frac[0][0] = 0.0
#         ud.tips.flux_frac[0][1] = 1.0
#         ud.tips.flux_frac[1][0] = 0.5
#         ud.tips.flux_frac[1][1] = 0.5

#         for k in range(ud.tips.NO_OF_RK_STAGES):
#             ud.tips.flux_frac[k,:] = 1e10

#         ud.tips.update_frac[0] = 0.5
#         ud.tips.update_frac[1] = 1.0

#         for k in range(ud.tips.NO_OF_RK_STAGES):
#             ud.tips.update_frac[k] = 1e10
        
#         ud.tips.multiD_updt = False
